update-go-version.sh

```diff
@@ -10,7 +10,7 @@ golang_base_version=$1
 echo "Updating golang version to $golang_base_version"
 
 go mod edit -go "${golang_base_version}.0"
-go mod edit -go "${golang_base_version}.0" tools/go.mod
+go mod edit -go "${golang_base_version}.1" tools/go.mod
```

Comment:
better to use `"${golang_base_version}.*"` or `"${golang_base_version}.[0-9]+"` so that we don't keep updating it for each minor release.